File: 2022/day1.nim
import sugar, math, algorithm
include prelude

var elves = "day1.txt".readFile.split("\n\n").map(elf => sum elf.split.map parseInt)
elves.sort

echo &"{elves[high(elves)]} {elves[high(elves)-2..high(elves)].sum}"

File: 2022/day1.py
from pathlib import Path

print(
  max(elves := sorted(
    map(lambda elf: sum(map(int, elf.split())),
        Path("day1.txt").read_text().split("\n\n")),
    reverse = True,
  )),
  sum(elves[:3]),
)
